flink-python/pyflink/table/types.py

```diff
@@ -16,171 +16,1811 @@
+        """
+        Constructs a RowType by adding new elements to it to define the schema. The method accepts
+        either:
+
+            a) A single parameter which is a RowField object.
+            b) 2 parameters as (name, data_type). The data_type parameter may be either a String
+               or a DataType object.
+
+        >>> row1 = RowType().add("f1", VarCharType()).add("f2", VarCharType())
+        >>> row2 = RowType([RowField("f1", VarCharType()), RowField("f2", VarCharType())])
+        >>> row1 == row2
+        True
+        >>> row1 = RowType().add(RowField("f1", VarCharType()))
+        >>> row2 = RowType([RowField("f1", VarCharType())])
+        >>> row1 == row2
+        True
+        >>> row1 = RowType().add("f1", "string")
+        >>> row2 = RowType([RowField("f1", VarCharType())])
+        >>> row1 == row2
+        True
+
+        :param field: Either the name of the field or a RowField object
+        :param data_type: If present, the DataType of the RowField to create
+        :return: a new updated RowType
+        """
+        if isinstance(field, RowField):
+            self.fields.append(field)
+            self.names.append(field.name)
+        else:
+            if isinstance(field, str) and data_type is None:
+                raise ValueError("Must specify DataType if passing name of row_field to create.")
+
+            if isinstance(data_type, str):
+                data_type_f = _parse_datatype_json_value(data_type)
+            else:
+                data_type_f = data_type
+            self.fields.append(RowField(field, data_type_f))
+            self.names.append(field)
+        # Precalculated list of fields that need conversion with
+        # from_sql_type/to_sql_type functions
+        self._need_conversion = [f.need_conversion() for f in self]
+        self._need_serialize_any_field = any(self._need_conversion)
+        return self
+
+    def __iter__(self):
+        """
+        Iterate the fields.
+        """
+        return iter(self.fields)
+
+    def __len__(self):
+        """
+        Returns the number of fields.
+        """
+        return len(self.fields)
+
+    def __getitem__(self, key):
+        """
+        Accesses fields by name or slice.
+        """
+        if isinstance(key, str):
+            for field in self:
+                if field.name == key:
+                    return field
+            raise KeyError('No RowField named {0}'.format(key))
+        elif isinstance(key, int):
+            try:
+                return self.fields[key]
+            except IndexError:
+                raise IndexError('RowType index out of range')
+        elif isinstance(key, slice):
+            return RowType(self.fields[key])
+        else:
+            raise TypeError('RowType keys should be strings, integers or slices')
+
+    def __repr__(self):
+        return "RowType(%s)" % ",".join(str(field) for field in self)
+
+    @classmethod
+    def from_json(cls, json):
```

Comment:
Add the convenient method is a pretty good idea, but for now, we do not add the `to_json`for pair, and this JIRA is the main goal is to align with JAVA API, So It's better to add it when suitable time. What do you think?

Reply:
Agree. We can remove `from_json` for now and add it when necessary.